axelrod/strategies/finite_state_machines.py

```diff
@@ -1,7 +1,220 @@
 from axelrod.action import Action
 from axelrod.player import Player
+from typing import DefaultDict, Iterator
+from collections import defaultdict, namedtuple
 
 C, D = Action.C, Action.D
+ALL_ACTIONS = [C, D]
+
+
+"""
+Memit = unit of memory.
+
+This represents the amount of memory that we gain with each new piece of
+history.  It includes a state, our_response that we make on our way into that
+state (in_act), and the opponent's action that makes us move out of that state
+(out_act).
+
+For example for this FSM:
+(0, C, 0, C),
+(0, D, 1, C),
+(1, C, 0, D),
+(1, D, 0, D)
+
+Has the memits:
+(C, 0, C),
+(C, 0, D),
+(D, 0, C),
+(D, 0, D),
+(C, 1, C),
+(C, 1, D)
+"""
+Memit = namedtuple("Memit", ["in_act", "state", "out_act"])
+
+def memits_match(x, y):
+    """In action and out actions are the same."""
+    return x.in_act == y.in_act and x.out_act == y.out_act
+def memit_sort(x, y):
+    """Returns a tuple of x in y, sorted so that (x, y) are viewed as the
+    same as (y, x).
+    """
+    if repr(x) <= repr(y):
+        return (x, y)
+    else:
+        return (y, x)
+
+
+def get_accessible_transitions(transitions: dict, initial_state: int) -> dict:
+  """Gets all transitions from the list that can be reached from the
+  initial_state.
+  """
+  edge_dict: DefaultDict[int, list] = defaultdict(list)
+  visited = dict()
+  for k, v in transitions.items():
```

Comment:
I changed k, v on the edge dictionaries.

I changed my "for k, v in transitions.items()" loops (this highlighted loop and others) to "for trans in transition_iterator(transitions)" as I had done in other spots.  This cleans it up.

However, I think it make sense to leave the k, v in the implementation of transition_iterator, because these don't have perfect meanings, and I'm kinda just moving around.